webargs/core.py

```diff
@@ -323,10 +323,12 @@ def _on_validation_error(self, error, req, schema):
         ):
             # Raise a webargs error instead
             kwargs = getattr(error, "kwargs", {})
-            kwargs["field_names"] = error.field_names
             kwargs["data"] = error.data
             if MARSHMALLOW_VERSION_INFO[0] < 3:
                 kwargs["fields"] = error.fields
+                kwargs["field_names"] = error.field_names
```

Comment:
It might be worth checking if the version is <3.0.0rc1 so that 3.0.0 beta users code won't be broken.

Reply:
Latest commit does this.